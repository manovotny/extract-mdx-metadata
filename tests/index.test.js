import {join} from 'path';

import {wrap} from 'jest-snapshot-serializer-raw';

import extractMdxMeta from '../index.js';

const getFixture = (name) => {
    const directory = new URL('.', import.meta.url).pathname;
    const path = join(directory, 'fixtures', `${name}.mdx`);

    return path;
};

test('should reset working directory when done', async () => {
    const expectedWorkingDirectory = process.cwd();
    const path = getFixture('static');

    await extractMdxMeta(path);

    expect(process.cwd()).toBe(expectedWorkingDirectory);

    expect().toMatchSnapshot();
});

test('should extract static meta', async () => {
    const path = getFixture('static');

    const meta = await extractMdxMeta(path);

    expect(wrap(meta)).toMatchSnapshot();
});

test('should extract dynamic meta', async () => {
    const path = getFixture('dynamic');

    const meta = await extractMdxMeta(path);

    expect(wrap(meta)).toMatchSnapshot();
});

test('should noop for any other export not named "meta"', async () => {
    const path = getFixture('other');

    const meta = await extractMdxMeta(path);

    expect(wrap(meta)).toMatchSnapshot();
});

test('should return build-in return value when no meta exists', async () => {
    const path = getFixture('none');

    const meta = await extractMdxMeta(path);

    expect(wrap(meta)).toMatchSnapshot();
});

test('should return custom return value when no meta exists', async () => {
    const options = {
        defaultReturnValue: undefined,
    };
    const path = getFixture('none');

    const meta = await extractMdxMeta(path, options);

    expect(wrap(meta)).toMatchSnapshot();
});
