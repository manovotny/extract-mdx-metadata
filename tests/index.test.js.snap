// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should extract dynamic meta 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): Object {
    "authors": Array [
      Object {
        "name": "Jane Doe",
        "twitter": "@janedoe",
        "website": "janedoe.com",
      },
      Object {
        "name": "John Doe",
        "twitter": "@johndoe",
        "website": "johndoe.com",
      },
    ],
    "date": "2020-01-01",
  },
}
`;

exports[`should extract static meta 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): Object {
    "prop": "value",
  },
}
`;

exports[`should noop for any other export not named "meta" 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): Object {},
}
`;

exports[`should reset working directory when done 1`] = `undefined`;

exports[`should return build-in return value when no meta exists 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): Object {},
}
`;

exports[`should return custom return value when no meta exists 1`] = `
Object {
  Symbol(jest-snapshot-serializer-raw): undefined,
}
`;
